src/app/views/query-runner/request/permissions/Permission.tsx

```diff
@@ -39,9 +40,10 @@ export class Permission extends Component<IPermissionProps, IPermissionState> {
   }
 
   public componentDidUpdate = (prevProps: IPermissionProps) => {
-    if ((prevProps.sample !== this.props.sample) || (prevProps.permissionsPanelOpen !== this.props.permissionsPanelOpen)) {
+    if ((prevProps.sample !== this.props.sample) || (prevProps.permissionsPanelOpen !== this.props.permissionsPanelOpen) || (prevProps.permissionModeType !== this.props.permissionModeType)) {
```

Comment:
```suggestion
    if (prevProps.sample !== this.props.sample 
            || prevProps.permissionsPanelOpen !== this.props.permissionsPanelOpen 
            || prevProps.permissionModeType !== this.props.permissionModeType) {
```
I think this line is a bit too long for general code cleanliness, since you have to scroll to read it.

Splitting the checks by the ||'s (or &&'s in other cases) is common practice to my knowledge and generally makes the code a lot cleaner/easier to read ^^

Reply:
yeah ill change that!

Reply:
resolved in https://github.com/LokiLabs/microsoft-graph-explorer-v4/pull/16/commits/55666db4d0964581599d5f907b355fb066034d67